numpy/core/tests/test_umath.py

```diff
@@ -2261,11 +2266,18 @@ def __array_prepare__(self, arr, context):
                 return np.array(arr).view(type=with_prepare)
 
         a = np.array(1).view(type=with_prepare)
-        x = np.add(a, a)
+        if use_where:
+            # Currently raises, due to the array being replaced during prepare
+            with pytest.raises(ValueError):
+                x = np.add(a, a, where=np.array(True))
+            return
+        else:
+            x = np.add(a, a)
```

Comment:
Is this a change in behavior?

Reply:
Well... It is a bug fix in a sense. Previously `__array_prepare__` wasn't called, which is why it didn't fail with where (although we never tested `where=`).